fix: add documentation to allow filtering on user additional information (#831)
gravitee-io/gravitee-docs: pages/am/3.x/user-guide/user-management/user/user-guide-users.adoc

NOTE: In AM Console, you do not need to specify the `filter` parameter, only the query value (for example `displayName eq "john doe"`).

WARNING: By default only the `id`, `externalId`, `username`, `email`, `displayName`, `firstName` and `lastName` fields are indexed in the database.

> This operation also supports searching the `additionalInformation` attributes of your users, but you need to create the correct indexes first. If you are using RDBMS backends, please refer to the underlying section `Add new filter fields for RDBMS backends` for more details

Reviewer: Missing period at end of sentence.
```suggestion
This operation also supports searching the `additionalInformation` attributes of your users, but you need to create the correct indexes first. If you are using RDBMS backends, please refer to the underlying section `Add new filter fields for RDBMS backends` for more details.
```

-> firstName co "john"
----

==== Add new filter field for RDBMS backends

> When you are using a RDMS backends, you will have to execute additional SQL statements to allow search on additionalInformation fields that are not managed by default by AM.

Reviewer: ```suggestion
When you are using a RDBMS backend, you will have to execute additional SQL statements to allow searching on additionalInformation fields that are not managed by default by AM.
```

> For PostgreSQL, as additional information is a JSON field you may have to create additional indexes.

Reviewer: Should we be consistent and use `additionalInformation`?

> For PostgreSQL, as additional information is a JSON field you may have to create additional indexes.

Reviewer: ```suggestion
For PostgreSQL, as `additionalInformation` is a JSON field you may have to create additional indexes.
```

[source, sql]
----
CREATE INDEX idx_users_additional_info_custom ON users USING BTREE (reference_id, reference_type, ((additional_information->>'custom_field')))
----

> For MySQL, MariaDB ans SQLServer, you will have to create a new column with the value coming from the user additional information and create an index.

Reviewer: ```suggestion
For MySQL, MariaDB and SQLServer, you will have to create a new column with the value coming from the user additional information and create an index.
```
